lib/flame_on/component/capture_schema.ex

```diff
@@ -0,0 +1,72 @@
+defmodule FlameOn.Component.CaptureSchema do
+  use Ecto.Schema
+  import Ecto.Changeset
+  alias Ecto.Changeset
+
+  schema "capture" do
+    field :module, :string
+    field :function, :string
+    field :arity, :integer
+    field :timeout, :integer
+  end
+
+  @default_attrs %{module: "cowboy_handler", function: "execute", arity: 2, timeout: 15000}
+
+  def changeset(attrs \\ @default_attrs) do
+    %__MODULE__{}
+    |> cast(attrs, [:module, :function, :arity, :timeout])
+    |> validate_required([:module, :function, :arity, :timeout])
+    |> validate_module()
+    |> validate_function_arity()
+  end
+
+  defp validate_module(%Changeset{valid?: false} = changeset), do: changeset
+
+  defp validate_module(changeset) do
+    module_str = get_field(changeset, :module)
+
+    module =
+      try do
+        String.to_existing_atom(module_str)
+      rescue
+        ArgumentError -> nil
+      end
+
+    if is_nil(module) or
+         not (function_exported?(module, :__info__, 1) or :erlang.check_old_code(module) or
+                :erlang.module_loaded(module)) do
+      if String.contains?(module_str, ".") and !String.starts_with?(module_str, "Elixir.") do
+        add_error(changeset, :module, "Elixir modules must start with \"Elixir.\"")
+      else
+        add_error(changeset, :module, "Module does not exist")
+      end
+    else
+      changeset
+    end
```

Comment:
Not a big deal but you could make this logic a bit more explicit:

```suggestion
    current_code_loaded = function_exported?(module, :__info__, 1) or :erlang.check_old_code(module) or :erlang.module_loaded(module)

    if module || current_code_loaded do
      changeset
    else
      if String.contains?(module_str, ".") and !String.starts_with?(module_str, "Elixir.") do
        add_error(changeset, :module, "Elixir modules must start with \"Elixir.\"")
      else
        add_error(changeset, :module, "Module does not exist")
      end
    end
```

Reply:
I'm going to leave this as is for now, it short circuits and avoids all the erlang calls when the module is not found (because `to_existing_atom` fails), which is going to be the majority case as users are typing in a module name. The "Elixir modules must start with "Elixir." is going away shortly anyway, so it will be cleaner.